deps/rabbit/src/rabbit_core_ff.erl

```diff
@@ -106,3 +115,43 @@ direct_exchange_routing_v2_migration(#ffcommand{name = FeatureName,
     end;
 direct_exchange_routing_v2_migration(_) ->
     ok.
+
+listener_records_in_ets_migration(#ffcommand{name = FeatureName,
+                                             command = enable}) ->
+    try
+        rabbit_misc:execute_mnesia_transaction(
+          fun () ->
+                  mnesia:lock({table, rabbit_listener}, read),
+                  Listeners = mnesia:select(rabbit_listener, [{'$1',[],['$1']}]),
+                  lists:foreach(
+                    fun(Listener) ->
+                            ets:insert(rabbit_listener_ets, Listener)
+                    end, Listeners)
+          end)
+    catch
+        throw:{error, {no_exists, rabbit_listener}} ->
+            ok;
+        throw:{error, Reason} ->
+            rabbit_log_feature_flags:error("Enabling feature flag ~s failed: ~p",
+                                           [FeatureName, Reason]),
+            {error, Reason}
+    end;
+listener_records_in_ets_migration(#ffcommand{name = FeatureName,
+                                             command = post_enable}) ->
+    try
+        case mnesia:delete_table(rabbit_listener) of
+            {atomic, ok} ->
+                ok;
+            {aborted, {no_exists, _}} ->
+                ok;
+            {aborted, Err} ->
+                rabbit_log_feature_flags:error("Enabling feature flag ~s failed to delete mnesia table: ~p",
+                                               [FeatureName, Err]),
+                {error, {failed_to_delete_mnesia_table, Err}}
+        end
+    catch
+        throw:{error, Reason} ->
+            rabbit_log_feature_flags:error("Enabling feature flag ~s failed: ~p",
+                                           [FeatureName, Reason]),
+            {error, Reason}
```

Comment:
I'm currently working on improving the callbacks and their spec, but the `post_enable` callback should always return `ok`. At that point, the feature flag is enabled and only non-criticals operations (like this cleanup) can happen at this stage.

The return value is ignored anyway, but this is just to make it clear that there is no rollback for instance.